src/app/cli/src/coda_main.ml

```diff
@@ -1033,6 +1033,31 @@ module Run (Config_in : Config_intf) (Program : Main_intf) = struct
           parent_hash previous ;
         Core.exit 1
 
+  module Payment_verifier =
+    Receipt_chain_database_lib.Verifier.Make (struct
+        include User_command
+
+        type payload = User_command.Payload.t
+
+        let payload {payload; _} = payload
```

Comment:
Please don’t monkey patch here. Can you either adjust the signature of the functor argument or adjust user command? I would suggest keeping payload as an inner module and adding this function to the real user command.